### PR TITLE
screamingfrogseospider: init at 12.4

### PR DESCRIPTION
**Pull request by mattmelling**
###### Motivation for this change
Adding package for Screaming Frog SEO Spider

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/screamingfrogseospider/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, fetchurl, dpkg, oraclejre }:
+with lib;
+let
+  version = "11.3";
+  url = "https://download.screamingfrog.co.uk/products/seo-spider/screamingfrogseospider_${version}_all.deb";
+  sha256 = "a6a7b9434dadcd0ca5d486df16386ce317ed7d4d2f7acc2730156e13859368f9";
+in stdenv.mkDerivation rec {
```
Comment (doronbehar) [inline pkgs/tools/misc/screamingfrogseospider/default.nix]: Why did you felt like you need the `let` + `in` syntax? Why not just use everything inside `mkDerivation`?

File: pkgs/tools/misc/screamingfrogseospider/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, fetchurl, dpkg, oraclejre }:
```
Comment (doronbehar) [inline pkgs/tools/misc/screamingfrogseospider/default.nix]: Just out of curiosity, other JREs don't work?

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/screamingfrogseospider/default.nix (hunk 1 above)
Comment (mattmelling) [reply pkgs/tools/misc/screamingfrogseospider/default.nix]: Honestly not sure, will push an update later to rectify this. Thanks for the feedback.

File: pkgs/tools/misc/screamingfrogseospider/default.nix (hunk 2 above)
Comment (mattmelling) [reply pkgs/tools/misc/screamingfrogseospider/default.nix]: At the time, only oraclejre and jetbrains jres had support for OpenJFX and the jetbrains jre straight up didn't work. IIRC at the time this was still in flux, so will review again given that it's been a few months.

**Comment by mattmelling** [comment]
I have addressed @doronbehar 's formatting comments. Left it using oraclejre as screamingfrog does not appear to be compatible with openjdk.

**Pull request closed without merging**